helix-term/src/ui/editor.rs

```diff
@@ -358,7 +361,16 @@ impl EditorView {
             let line_number_text = if line == last_line && !draw_last {
                 "    ~".into()
             } else {
-                format!("{:>5}", line + 1)
+                match config.line_number {
+                    LineNumber::Absolute => format!("{:>5}", line + 1),
+                    LineNumber::Relative => {
+                        // TODO: Put char_to_line out of the loop
+                        let current_line = doc.text().char_to_line(selections.primary().anchor);
```

Comment:
You should extract this outside of the loop, should be fine to just place it immediately before.